jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -335,8 +335,15 @@ class DeploymentFlow extends React.Component {
       credential: this.state.credential,
       region: this.state.region
     };
-    if (this.state.sshKey) {
-      args.config['authorized-keys'] = this.state.sshKey;
+    if (this.state.sshKeys.length) {
+      // Attempt to provide at least one key in case the addSSHKeys call fails.
+      args.config['authorized-keys'] = this.state.sshKeys[0].text;
```

Comment:
I explained on the call that we need this in case the entire call addKeys fails. Yes, adding individual keys is a no-op, but adding a series of keys may fail for other reasons.